src/test/com/money_account_service/utility/UserServiceClientTestUT.groovy

```diff
@@ -0,0 +1,58 @@
+package com.money_account_service.utility
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import okhttp3.*
+import org.json.JSONObject
+import org.spockframework.spring.SpringBean
+import org.springframework.test.util.ReflectionTestUtils
+import spock.lang.Specification
+
+class UserServiceClientTestUT extends Specification {
```

Comment:
What do you want to achieve with this test? It's hard for me to understand the setup with mixed mocks, actual clients, request/response bodies.

For testing REST clients, usually https://wiremock.org/ can be used, to stub the HTTP server. Also, mocking or using a client like `OkHttpClient` is usually a bad idea, because you hide a lot of complexity.

Unit tests are good to verify some behavior is onvoked, but on the http client level it seems too confusing. If you want to test this client, I would opt in favour of more integration like test, with WireMock.